[Generic SNMP Polling Plugin] SNMP Plugin Config Incorrect Type

See the following two functions. If the user attempts to override SNMP function arguments by adding non_repeaters or max_repetitions to the .panoptes-plugin config, it will not work. Since Yapsy doesn't allow us to use a configspec file while parsing the config, it defaults to interpreting every value as a string. Interpreting the SNMP vars as strings causes the following issue when sent to Steamroller proxy hosts.

 [ERROR] One or more phases invalid: Invalid options "{\'non_repeaters\': 0, \'max_repetitions\': \'20\'}"')

File: yahoo_panoptes/plugins/polling/generic/snmp/plugin_polling_generic_snmp.py
```python3
def _get_snmp_polling_var(self, var, default):
        if self._config.get(u'snmp'):
            if var in self._config[u'snmp']:
                return self._config[u'snmp'].get(var)

        if u'snmp' in self._plugin_context.config:
            if var in self._plugin_context.config[u'snmp']:
                return self._plugin_context.config[u'snmp'].get(var)

        return default

def _build_map_by_bulk_walk(self, oid_name):
        """
        Builds the oids_map for the provided oid_name using snmp bulk walk.
        Args:
            oid_name: The oid for which to build the map using snmp bulk walk.
        Returns:
            None
        """
        self._oid_maps[oid_name] = {}
        device_metrics_map = dict()
        stats = None
        try:
            if self._config[u"oids"][oid_name][u"method"] == u"bulk_walk":
                stats = self._snmp_connection.bulk_walk(oid=self._config[u"oids"][oid_name][u"oid"],
                                                        non_repeaters=self._get_snmp_polling_var(
                                                            u"non_repeaters", 0),
                                                        max_repetitions=self._get_snmp_polling_var(
                                                            u"max_repetitions", _MAX_REPETITIONS))
   ```
   
   Fix: Cast the result from the self._get_snmp_polling_var() function when it is passed either 'non_repeaters' or 'max_repetitions'.

--- a/yahoo_panoptes/plugins/polling/generic/snmp/plugin_polling_generic_snmp.py
+++ b/yahoo_panoptes/plugins/polling/generic/snmp/plugin_polling_generic_snmp.py
@@ -190,10 +190,10 @@ class PluginPollingGenericSNMPMetrics(polling_plugin.PanoptesPollingPlugin):
         try:
             if self._config[u"oids"][oid_name][u"method"] == u"bulk_walk":
                 stats = self._snmp_connection.bulk_walk(oid=self._config[u"oids"][oid_name][u"oid"],
-                                                        non_repeaters=self._get_snmp_polling_var(
-                                                            u"non_repeaters", 0),
-                                                        max_repetitions=self._get_snmp_polling_var(
-                                                            u"max_repetitions", _MAX_REPETITIONS))
+                                                        non_repeaters=int(self._get_snmp_polling_var(
+                                                            u"non_repeaters", 0)),
+                                                        max_repetitions=int(self._get_snmp_polling_var(
+                                                            u"max_repetitions", _MAX_REPETITIONS)))
         except Exception as e:
             self._polling_status.handle_exception(u"device", e)
             self._handle_exceptions_for_oid(oid_name, e)

